fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/domain/SearchQuery.java

```diff
@@ -76,11 +76,8 @@ public void add(SearchExtension ext) {
         T query = getRoot(visitor);
         visitExtensions(query, visitor);
 
-        // Add the parameters subquery and add the extensions there too
+        // Add the parameters subquery
         T parameterBase = visitor.getParameterBaseQuery(query);
-        for (SearchExtension ext: this.extensions) {
```

Comment:
But what about the extensions used by other query types?

Reply:
we still visit the extensions and add them to the outer query.
we had added these three lines (81-83) just to help postgresql choose a more efficient plan in the case of the WholeSystemFilterQuery.  since it caused unanticipated regression on other queries (e.g. the count), i've moved this to the specific subtype where we were seeing the original issue.